src/extension/experimentService/experimentService.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Configstore from "configstore";
+import * as https from "https";
+import { PreviewVersionPromotion} from "./experiments/previewVersionPromotion";
+import { TelemetryHelper } from "../../common/telemetryHelper";
+
+export enum ExperimentStatuses {
+    FAILURE = "failure",
+    SUCCESS = "success",
+    SKIPPED = "skipped",
+}
+
+export interface ExperimentConfig {
+    experimentName: string;
+    popCoveragePercent: number;
+    enabled: boolean;
+}
+
+export interface ExperimentParameters extends ExperimentConfig {
+    [key: string]: any;
+    extensionId?: string;
+}
+
+export interface ExperimentResult {
+    resultStatus: ExperimentStatuses;
+    updatedExperimentParameters: ExperimentParameters;
+}
+
+export class ExperimentService {
+    private readonly endpointURL: string;
+    private readonly configName: string;
+    private readonly availableExperiments: any;
+    private config: Configstore;
+    private downloadedExperimentsConfig: Array<ExperimentConfig> | null;
+    private downloadConfigProm: Promise<ExperimentConfig[]> | null;
+
+    constructor() {
+        this.endpointURL = "https://microsoft.github.io/vscode-react-native/experiments/experimentsConfig.json";
+        this.configName = "reactNativeToolsConfig";
+        this.availableExperiments = {
+            RNT_PREVIEW_PROMPT: "RNTPreviewPrompt",
+        };
+
+        this.config = new Configstore(this.configName);
+        this.downloadedExperimentsConfig = null;
+        this.downloadConfigProm = null;
+    }
+
+    public async initialize(): Promise<void> {
+        this.downloadConfigProm = this.downloadExperimentsConfig();
+    }
+
+    public async runExperiments(): Promise<void> {
+        if (!this.downloadedExperimentsConfig) {
+            if (!this.downloadConfigProm) {
+                throw new Error("Experiment Service is not initialized");
+            }
+            try {
+                this.downloadedExperimentsConfig = await this.downloadConfigProm;
+            } catch (err) {
+                this.downloadConfigProm = this.downloadExperimentsConfig();
+                throw new Error("Failed to download experiments config");
```

Comment:
Should these strings be localized?

Reply:
I think that users shouldn't see errors from the Experiment Service and we should print such errors in `verbose` mode.